### PR TITLE
Reliably handle Offer -> Checkout navigation scenario by creating new priceIntent if presious one was used to add offer to cart

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Reimplement logic for clearing priceIntent after it was used for adding item to cart

Old solution did not work reliably since product -> checkout navigation crossed app router -> pages router barrier and was a full page reload

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4405?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4405** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4405?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4404** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4404?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by gustaveen — APPROVED** [review]
LGTM! 👌

File: apps/store/src/components/ProductPage/PriceIntentContext.tsx [hunk 1]
```diff
@@ -38,34 +38,51 @@ const usePriceIntentContextValue = () => {
...
-        shopSessionId: shopSession.id,
+        shopSessionId,
       })
+      // Use new priceIntent if
+      //  - one of current one's offers are added to cart
+      //  - and we're not in 'edit cart item' mode
+      // This may happen during app -> pages router navigation when using "go to checkout" button in purchase form
+      const cartPriceIntentIds = new Set(cartRef.current?.entries.map((item) => item.priceIntentId))
+      if (entryToReplace == null && cartPriceIntentIds.has(priceIntent.id)) {
+        console.log('Current priceIntent used in one of cart items, creating new ones')
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductPage/PriceIntentContext.tsx:69]: nit
```suggestion
        console.log('Current priceIntent used in one of cart items, creating new one')
```

**Review by guilhermespopolin — APPROVED** [review]
Nice :)

**Comment by alebedev** [comment]
### Merge activity

* **Jun 26, 7:47 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4405) rebased this pull request as part of a merge.

**Pull request merged**